apps/src/blockly/addons/cdoUtils.js

```diff
@@ -456,3 +459,30 @@ export function appendSharedFunctions(startBlocksSource, functionsXml) {
   }
   return startBlocks;
 }
+/**
+ * Update the XML string representing toolbox data for compatibility with
+ * Google Blockly.
+ * This function potentially modifies each <block> element in the XML
+ * if there are unsupported attributes, missing mutators, etc.
+ * We also process block xml during domToBlockSpace, which is called to
+ * convert sources from XML to JSON.
+ *
+ * @param {string} toolboxString - The XML string representing toolbox data.
+ * @returns {string} The modified XML string after processing.
+ */
+export function processToolboxXml(toolboxString) {
+  const parser = new DOMParser();
+  const xmlDoc = parser.parseFromString(toolboxString, 'text/xml');
+  const xmlRoot = xmlDoc.documentElement;
+  const blocks = Array.from(xmlRoot.childNodes).filter(
+    node => node.nodeName === 'block'
+  );
+
+  blocks.forEach(block => {
+    processIndividualBlock(block);
+  });
+
```

Comment:
Do you need to do the array conversion? Not sure if something more direct like this would work

```
    const blocks = xmlDoc.querySelectorAll('block'); // Simplify node filtering
    blocks.forEach(processIndividualBlock); // Process each block directly
```

Reply:
Much less verbose! Thank you. :)